test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/PropertiesTest.scala

```diff
@@ -173,4 +175,286 @@ class PropertiesTest {
+    prop.load(is)
+    is.close()
+    checkLoadFromFile(prop)
+  }
+
+  @Test def loadReader(): Unit = {
+    val prop = loadReader(dummyProps)
+
+    assertEquals("value1", prop.getProperty("key1"))
+    assertNull(prop.getProperty("commented.key"))
+    assertEquals("default_value",
+                 prop.getProperty("commented.key", "default_value"))
+  }
+
+  @Test def loadReaderHandleSpecialChars(): Unit = {
+    val prop = loadReader("#\u008d\u00d2\na=\u008d\u00d3")
+    assertEquals("\u008d\u00d3", prop.get("a"))
+
+    val prop1 = loadReader("#properties file\r\nfred=1\r\n#last comment")
+    assertEquals("1", prop1.get("fred"))
+  }
+
+  @Test def loadReaderWithFileInput(): Unit = {
+    // string input for Scala.js
+    val is = new ByteArrayInputStream(filestr.getBytes())
+    val prop = new Properties()
+    prop.load(new InputStreamReader(is))
+    is.close()
+    checkLoadFromFile(prop)
+  }
+
+  @Test def storeOutputStreamCommentsLoadInputStreamRoundtrip(): Unit = {
+    val prop1 = new Properties()
+    prop1.put("Property A", " aye\\\f\t\n\r\b")
+    prop1.put("Property B", "b ee#!=:")
+    prop1.put("Property C", "see")
+    val header1 =
+      "A Header\rLine2\nLine3\r\nLine4\n!AfterExclaim\r\n#AfterPound\nWow!"
+    val out1 = storeStream(prop1, header1)
+
+    val prop2 = loadByteArrayOutputStream(out1)
+    assertAll(expected = prop1, actual = prop2)
+    // Avoid variable Date output which is last line in comment
+    // Matches JVM output
+    val commentsWithoutDate =
+      """|#A Header
+         |#Line2
+         |#Line3
+         |#Line4
+         |!AfterExclaim
+         |#AfterPound
+         |#Wow!""".stripMargin
+    assertTrue(out1.toString().startsWith(commentsWithoutDate))
+  }
+
+  @Test def checkPropertiesFormattedCorrectly(): Unit = {
+    val prop1 = new Properties()
+    prop1.put("Property C", "see")
+    val out1 = storeStream(prop1)
+    assertTrue(out1.toString().trim().endsWith("""Property\ C=see"""))
+
+    val prop2 = new Properties()
+    prop2.put("Property B", "b ee#!=:")
+    val out2 = storeStream(prop2)
+    assertTrue(out2.toString().trim().endsWith("""Property\ B=b ee\#\!\=\:"""))
+
+    val prop3 = new Properties()
+    prop3.put("Property A", " aye\\\f\t\n\r\b")
+    val out3 = storeStream(prop3)
+    // JVM outputs \b as \u0008 and you can't just add \u0008 to the end
+    val result = new StringBuilder("""Property\ A=\ aye\\\f\t\n\r""")
+      .appendAll(Array('\\', 'u', '0', '0', '0', '8'))
+    assertTrue(out3.toString().trim().endsWith(result.toString))
+  }
+
+  @Test def storeWriterComments(): Unit = {
+    val prop1 = new Properties()
+    prop1.put("Property A", " aye\\\f\t\n\r\b")
+    prop1.put("Property B", "b ee#!=:")
+    prop1.put("Property C", "see")
+    val out = storeWriter(prop1, "A Header")
+    val prop2 = loadByteArrayOutputStream(out)
+    assertAll(prop1, prop2)
+  }
+
+  // helper functions
+
+  def storeStream(props: Properties,
```

Comment:
I think these various alignments should simply be 4 spaces vs aligned with first param (e.g. https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#class-declaration and https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#indentation

Reply:
These too.